main.cpp

```diff
@@ -107,87 +120,62 @@ struct Numeric
 {
     using Type = Temporary<T>;
 
+    // rule of 5: define destructor 
+    ~Numeric() = default; // std::unique_ptr<Type> does not require management
+
+    // rule of 5: define move constructor
+    Numeric (Numeric&& other) : value (std::move (other.value)) 
+    {
+        other.value = nullptr;
+    }
+
+    // rule of 5: define move assignment operator
+    Numeric& operator= (Numeric&& other)
+    {
+        value = std::move (other.value);
+        other.value = nullptr;
+        return *this;
+    }
+
+    // added to suppress an error when assigning a const T to a Numeric<T>
     Numeric& operator= (const T& rhs)
     {
         *value = rhs;
         return *this;
     }
 
```

Comment:
This is unneeded if you change the `Numeric(Type t)` constructor. 
hint: `Type` is what should be changed
- [ ] revise